middleware_defaults.go

```diff
@@ -25,6 +25,16 @@ func (d *JobInsertMiddlewareDefaults) InsertMany(ctx context.Context, manyParams
 	return doInner(ctx)
 }
 
+// JobInsertMiddlewareFunc is a convenience helper for implementing
+// rivertype.JobInsertMiddleware using a simple function instead of a struct.
+type JobInsertMiddlewareFunc func(ctx context.Context, manyParams []*rivertype.JobInsertParams, doInner func(ctx context.Context) ([]*rivertype.JobInsertResult, error)) ([]*rivertype.JobInsertResult, error)
```

Comment:
Added func helpers for middlewares because it makes it really easy to isolate middlewares to specific test cases, even if the middleware needs to embed something like `baseservice.BaseService`.